uefi-raw/src/protocol/device_path.rs

```diff
@@ -20,6 +76,946 @@ impl DevicePathProtocol {
     pub const GUID: Guid = guid!("09576e91-6d3f-11d2-8e39-00a0c969723b");
 }
 
+/// Hardware Device Path Type.
+pub const HARDWARE_DEVICE_PATH: u8 = 0x01;
+
+/// PCI Device Path SubType.
+pub const HW_PCI_DP: u8 = 0x01;
+
+/// PCI Device Path.
+#[derive(Debug)]
+#[repr(C,packed)]
```

Comment:
```suggestion
#[repr(C)]
```

Same here. Add a unit test that tests the ABI guarantees. Please only use `packed` if the default layout is wrong